File: API2/Entities/Product.cs
using System.ComponentModel.DataAnnotations;

namespace API2.Entities
{
    public class Product
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public long Price { get; set; }
        public string PictureUrl { get; set; }
        public string PictureType { get; set; }
        public string Brand { get; set; }
        public int QuantityInStock { get; set; }
    }
}

File: storeui/src/app/models/Product.ts
export interface Product{
  id: number
  name: string
  description: string
  price: number
  pictureUrl: string
  pictureType?: string
  brand?: string
  quantityInStock?: number
}

export interface ProductParams{
  orderBy : string,
  searchTerm? :string,
  brand? :string[],
  type? :string[],
  pageNumber : number,
  pageSize:number
}